src/show_message.rs

```diff
@@ -0,0 +1,137 @@
+use std::borrow::Cow;
+
+use itertools::Itertools;
+use jsonrpc_core::{Id, MethodCall};
+use lsp_types::{MessageActionItem, MessageType, ShowMessageParams, ShowMessageRequestParams};
+use serde::Deserialize;
+
+use crate::{context::Context, types::EditorMeta, util::editor_quote};
+
+/// Queues the message request from the LSP server.
+pub fn show_message_request(meta: EditorMeta, request: MethodCall, ctx: &mut Context) {
+    let params: ShowMessageRequestParams = request
+        .params
+        .clone()
```

Comment:
don't clone the request here, rather use `let request_id = request.id` or similar